### DISCUSSION TITLE
Get data attributes on click element

### DISCUSSION THREAD

**Discussion by MuhammadSaim** [alpinejs/alpine#1288] [1. Help] [answered]
I have an element in alpinejs I want to get values of the data attributes when I click on that link.

```html
<li class="">
    <a @click.prevent="download()" data-video="video_id" data-url="{{ base_url('tools/download_thumbnail') }}" data-quality="maxres"  class="rounded-b bg-white hover:bg-public-primary hover:text-white  py-2 px-4 block whitespace-no-wrap" href="#">Max Res</a>
</li>
```

![image](https://user-images.githubusercontent.com/19898499/116805305-dda78f00-ab3e-11eb-84e0-99c8a8422bfc.png)

**Comment by KevinBatdorf** [comment, marked as answer]
Can't see your `x-data` but maybe you are looking for `$event` instead of `e`?

**Reply by MuhammadSaim** [reply]
I don't want to get the x-data I wanna value of the HTML data attributes ```data-quantity``` when I click on the element.

**Reply by SimoTod** [reply]
You are passing `e` to a function call but that variable doesn't exist (unless defined in x-data but I assume it is not). As Kevin suggested, you are probably after `download($event)`.
